### PR TITLE
Make 'compass list --suites -v' print the contents of a suite

### PR DESCRIPTION
**Pull request by matthewhoffman**
This merge also updates the documentation for this change and adds a
short section to the Developer's Guide about developing suites.

### CODE REVIEW

File: compass/list.py [hunk 1]
```diff
@@ -21,6 +22,7 @@ def list_cases(test_expr=None, number=None, verbose=False):
 
     verbose : bool, optional
         Whether to print details of each test or just the subdirectories
+        When applied to suites, verbose will list the tests in the suite
```
Comment (xylar) [inline compass/list.py:25]:
```suggestion
        when applied to suites, verbose will list the tests in the suite
```

Comment (xylar) [inline compass/list.py:25]:
```suggestion
        Whether to print details of each test or just the subdirectories.
        When applied to suites, verbose will list the tests in the suite.
```
If you have multiple sentences, the API documentation may put them on the same line (the line breaks here aren't necessarily retained) so it's important to punctuate.

File: compass/list.py [hunk 2]
```diff
@@ -120,7 +128,7 @@ def main():
     if args.machines:
         list_machines()
     elif args.suites:
-        list_suites()
+        list_suites(verbose=args.verbose)
```
Comment (xylar) [inline compass/list.py:131]: Could you update the help string for `--verbose` to also mention suites?

File: docs/developers_guide/organization.rst [hunk 3]
```diff
@@ -1718,3 +1718,16 @@ forgotten to compile the model, this will be obvious by the broken symlink and
...
+
+Test Suites
+-----------
+
+As described in the :ref:`user_testsuites` section of the User's Guide, COMPASS
+test cases can be organized into test suites.  Each core has separate regression
+suites, and a core can have multiple independent regression suites.  A developer
+defines a test suite by creating a `.txt` file within the `compass/CORE/suites`
+directory.  The format of the `.txt` file is a list of the paths to the tests
+desired to be part of the suite.  A line starting with `#` will be treated as a
```
Comment (xylar) [inline docs/developers_guide/organization.rst:1732]:
```suggestion
directory.  The format of the `.txt` file is a list of the work directories of the tests
desired to be part of the suite.  A line starting with `#` will be treated as a
```

File: compass/list.py [hunk 4]
```diff
@@ -97,7 +99,13 @@ def list_suites(cores=None):
         for suite in sorted(suites):
             if suite.endswith('.txt'):
                 print('  -c {} -t {}'.format(core, os.path.splitext(suite)[0]))
-
+                if verbose:
+                   text = resources.read_text('compass.{}.suites'.format(core), suite)
```
Comment (xylar) [inline compass/list.py:103]:
```suggestion
                   text = resources.read_text('compass.{}.suites'.format(core), 
                                              suite)
```

File: compass/list.py [hunk 5]
```diff
@@ -97,7 +99,13 @@ def list_suites(cores=None):
...
+                   tests = list()
+                   for test in text.split('\n'):
+                      test = test.strip()
+                      if len(test) > 0 and test not in tests and not test.startswith('#'):
+                         print("\t* {}".format(test))
```
Comment (xylar) [inline compass/list.py:108]:
```suggestion
                      if len(test) > 0 and test not in tests and \
                              not test.startswith('#'):
                          print("\t* {}".format(test))
```

**Review by xylar — CHANGES_REQUESTED** [review]
This is a really great addition and works like a charm.  I hadn't originally done this because the text files are pretty easy to read but I can definitely see the benefit because they're not that easy to find.  This will be really handy.

I've made a few small suggested edits.  In addition, it would be great if you could take care of the 8 linting issues found above by pep8speaks.  I would like us to stick with the PEP8 style guide pretty strictly.  The main exception are long lines that really can't be broken up without hampering readability, such as very long type names or URLs in doc strings.  Please make sure indentation is always a multiple of 4.  The two long lines pointed out here could be broken into multiple lines as I have suggested below.

**Comment by xylar** [comment]
@matthewhoffman, I also really appreciate you taking the initiative to implement this, and not just request this feature.  It makes a really big difference to how the project feels to have others diving into development.

File: compass/list.py (hunk 5 above)
Comment (xylar) [reply compass/list.py:108]: One more thing. I'm not sure why pep8speaks didn't point it out but there need to be 2 blank lines between functions (at line 109).

**Comment by matthewhoffman** [comment]
@xylar , thanks for reviewing this and providing me feedback on coding style and documentation.  I have flake8 installed now (including a vim plugin), and it really makes linting easy.  (flake8 did identify the missing 2 blank lines after a function, BTW.)  I was aware there were coding standards but I wasn't sure how to easily check my code against them so I kinda ignored it the first time around. :)  I've made all the changes you requested - let me know if you see anything else.

I realized this was a kind of trivial PR to make, functionality-wise, but I had a few reasons to go ahead with it.  The first was I was trying to figure out if I could comment my suite definition .txt files, and this seemed like a nice way to test that.  The second was that I realized that for a user using the released python package, they would not have access to the .txt files (or even for someone working from the repo, they might not be familiar with where to find them).  And then I decided this was a nice simple modification for me to practice making a compass PR!   It's nice to get my feet wet and feel like I can jump into the framework.  With the developer docs and the well-commented source code, it was really easy to figure out how to implement this.

**Comment by matthewhoffman** [comment]
@xylar , hold of on re-reviewing.  I think some changes are required to support comments in the suite .txt files.  I had tested that yesterday, but I most of have missed something or had a problem with my conda env.

**Comment by xylar** [comment]
> I think some changes are required to support comments in the suite .txt files.

Yes, I also just tested and all values are parsed, so you are right.  It should be pretty easy to add comment parsing.

**Comment by matthewhoffman** [comment]
I added a new commit to support test case commenting by the `compass suite` command.  I had previously added it for `compass list` but forgot to also add it to `compass suite`.  

I tested by adding a new line to the landice integaration suite:
```
# foo
```
and I confirmed it was ignored by:
* `compass list --suites -v`
* `compass suite -c landice -t sia_integration -w $scratch4/compass_testing -s -m badger`
* `compass suite -c landice -t sia_integration -w $scratch4/compass_testing --clean -m badger`

The comment line was ignored by all those commands.  This is ready for final review and merge.

**Review by xylar — APPROVED** [review]
@matthewhoffman this looks good.  I think your testing is sufficient.  Feel free to merge when you're ready

**Pull request merged**